lib/ansible/vars/hostvars.py

```diff
@@ -85,7 +86,7 @@ def raw_get(self, host_name):
 
     def __getitem__(self, host_name):
         data = self.raw_get(host_name)
-        sha1_hash = sha1(str(data).encode('utf-8')).hexdigest()
+        sha1_hash = sha1(to_text(data, errors='surrogate_or_replace').encode('utf-8')).hexdigest()
```

Comment:
Heh.  Since this code is then transforming the text string into bytes, just do this instead:
``` python
from ansible.module_utils._text import to_bytes
[...]
sha1_hash = sha1(to_bytes(data, errors='surrogate_or_strict')).hexdigest()
```